moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -319,26 +321,31 @@ void RobotTrajectory::getRobotTrajectoryMsg(moveit_msgs::msg::RobotTrajectory& t
         }
       }
       if (duration_from_previous_.size() > i)
-        trajectory.multi_dof_joint_trajectory.points[i].time_from_start = ros::Duration(total_time);
+        trajectory.multi_dof_joint_trajectory.points[i].time_from_start = dur_total;
       else
         trajectory.multi_dof_joint_trajectory.points[i].time_from_start = ZERO_DURATION;
     }
   }
 }
 
 void RobotTrajectory::setRobotTrajectoryMsg(const robot_state::RobotState& reference_state,
-                                            const trajectory_msgs::JointTrajectory& trajectory)
+                                            const trajectory_msgs::msg::JointTrajectory& trajectory)
 {
   // make a copy just in case the next clear() removes the memory for the reference passed in
+
   const robot_state::RobotState& copy = reference_state;
   clear();
   std::size_t state_count = trajectory.points.size();
-  ros::Time last_time_stamp = trajectory.header.stamp;
-  ros::Time this_time_stamp = last_time_stamp;
+  rclcpp::Time last_time_stamp = trajectory.header.stamp;
+  rclcpp::Time this_time_stamp = last_time_stamp;
+
+  rclcpp::Time trajStamp = trajectory.header.stamp;
+  rclcpp::Duration dur_from_start(0, 0);
 
   for (std::size_t i = 0; i < state_count; ++i)
   {
-    this_time_stamp = trajectory.header.stamp + trajectory.points[i].time_from_start;
+    this_time_stamp = trajStamp + rclcpp::Duration(trajectory.points[i].time_from_start.sec,
```

Comment:
Isn't `time_from_start` already an `rclcpp::Duration` so we can just add it directly?